Update GDScript code style guide (#265)
GDQuest/website: content/docs/guidelines/best-practices/godot-gdscript/_index.md

```

{{% notice note %}}

> Starting from Godot 3.2, you can write docstrings as a series of comments above any class, property, signal, or function. The GDScript language server will show the comments as part of the description in the code completion box. You can also use that to create a code reference with our tool [GDScript Docs Maker](https://github.com/GDQuest/gdscript-docs-maker).

Author: ```suggestion
Starting from Godot 3.2, you can write docstrings as a series of comments above any class, property, signal, or function. The GDScript language server will show the comments as part of the description in the docs preview, if your code editor supports this feature. You can also use that to create a code reference with our tool [GDScript Docs Maker](https://github.com/GDQuest/gdscript-docs-maker).
```

Author: This is specifically a docs preview feature that works over the language server, separate from auto-completion.

```gdscript
func drag_to(event_position: Vector2) -> void:
	# Calculate the position of the mouse cursor relative to the RectExtents' center
	var viewport_transform_inverse:= rect_extents.get_viewport().get_global_canvas_transform().affine_inverse()
	var viewport_position: Vector2 = viewport_transform_inv.xform(event_position)
	var transform_inverse:= rect_extents.get_global_transform().affine_inverse()
	var target_position: Vector2 = transform_inv.xform(viewport_position.round())
```

In the following example, the comment explains why a seemingly strange line of code is necessary. This ensures another developer doesn't remove it accidentally:

```gdscript
extends BattlerAI

func choose_action(actor: Battler, targets: Array = []):
	# We use yield even though the action is instantaneous
	# because the combat system expects this method to use a coroutine
	yield(get_tree(), "idle_frame")
	#...
```

## File naming and folder conventions

We follow some guidelines to keep the name of our files meaningful and consistent.

We arrange and name our files in a way that reflects and informs us on the project's code structure.

In short, the rules are:

> - Name directories, scenes, and script files in `PascalCase`.

Author: ```suggestion
- Name directories, scenes, and script files in `PascalCase`.
- Name scripts and scenes after a scene's root node.
```

![](images/names-filesystem.png)

Always name your script and scene after the scene's root node if applicable. This makes it much easier to search for the files corresponding to a scene.

> <!-- "Name scripts and scenes after a scene's root node", should probably be included in the top list -->

Author: ```suggestion
```
